core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,277 @@
+        backdrop: 'static',
+        scope: $scope,
+        controller: [
+          '$scope', '$rootScope', '$modalInstance', 'editorContextService',
+          function($scope, $rootScope, $modalInstance, editorContextService) {
+            $scope.EXPLANATION_FORM_SCHEMA = {
+              type: 'html',
+              ui_config: {}
+            };
+
+            $scope.INTERACTION_SPECS = INTERACTION_SPECS;
+
+            $scope.tmpSolution = {};
+            $scope.tmpSolution.answerIsExclusive = false;
+            $scope.tmpSolution.correctAnswer = '';
+            $scope.tmpSolution.explanation = '';
+
+            $scope.addSolutionForm = {};
+
+            if ($scope.objectType === 'CodeString') {
+              stateSolutionService.displayed = {
+                correctAnswer: {
+                  code: '',
+                  output: '',
+                  evaluation: '',
+                  error: ''
+                }
+              };
+            }
+
+            $scope.saveSolution = function() {
+              // Close the modal and save it afterwards.
+              var answer = ($scope.correctAnswer !== null) ? (
+                $scope.correctAnswer) : $scope.tmpSolution.correctAnswer;
+              if ($scope.objectType === 'CodeString') {
+                answer = {
+                  code: $scope.tmpSolution.correctAnswer,
+                  output: '',
+                  evaluation: '',
+                  error: ''
+                };
+              } else if ($scope.objectType === 'UnicodeString') {
+                answer = {
+                  ascii: '',
+                  latex: $scope.tmpSolution.correctAnswer
+                };
+              }
+
+              $modalInstance.close({
+                solution: angular.copy(
+                  SolutionObjectFactory.createNew(
+                    $scope.tmpSolution.answerIsExclusive,
+                    answer,
+                    $scope.tmpSolution.explanation))
+              });
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function(result) {
+        var explorationId =
+          explorationContextService.getExplorationId();
+        var currentStateName =
+          editorContextService.getActiveStateName();
+        var state = explorationStatesService.getState(currentStateName);
+        var interactionId = stateInteractionIdService.savedMemento;
+        var rulesServiceName = (
+          angularNameService.getNameOfInteractionRulesService(interactionId));
+        // Inject RulesService dynamically.
+        var rulesService = $injector.get(rulesServiceName);
+        var correctAnswer = result.solution.correctAnswer;
+        try {
+          AnswerClassificationService.getMatchingClassificationResult(
+            explorationId, state, correctAnswer, true, rulesService)
```

Comment:
Move the ")" to the next line, so you have:

```
          AnswerClassificationService.getMatchingClassificationResult(
            explorationId, state, correctAnswer, true, rulesService
          ).then(function(result) {
            ...
          });
```

Easier to see the matching parens that way.